### ISSUE TITLE
[Q] User-defined state/command descriptions with metadata

### ISSUE THREAD

**Issue by ghys** [openhab/openhab-core#1185]
AFAIK (correct me if I'm wrong) there is no way for an user to set/override the state description or command description of a managed item, for instance specifying a custom format pattern between brackets in the label like DSL items.
The problem is that sometimes, the state/command descriptions provided by the channel might have to be altered, or the item might not be linked at all and managed solely with rules.

Would it make sense to have a `StateDescriptionProvider` resp. `CommandDescriptionProvider` reading the configuration parameters in well-known metadata namespaces - say, _stateDescription_ & _commandDescription_?

It could maybe even work for DSL items as well e.g.
```
Number EnergyCost { stateDescription="???" [ pattern="%.1f €", readOnly=true ] }
Switch VideoInput { commandDescription="???" [ options="HDMI1,HDMI2" ] }
```

**Comment by openhab-5iver** [comment]
> AFAIK (correct me if I'm wrong) there is no way for an user to set/override the state description or command description of a managed item

I played around with this for a while using scriupted automation and the only way I saw to do it was to create a Thing with a Channel that the Item was linked to in order to manipulate the Item's StateDescription. I was planning to make a helper library for this, but it was getting so involved that I stopped, since OHC would be the right place to do it.

This functionality would be really nice to have. Maybe something for the OH 3.0 backlog, @cweitkamp?
